src/hooks/useActiveProject.tsx

```diff
@@ -6,164 +6,87 @@ import React, {
-      console.warn('Ignoring attempt to set invalid projectId', projectId);
-    }
-    return getDefault();
-  }
-  return { selectedProject, selectedSubject };
-};
-
 export const ActiveProjectContextProvider = ({
   children,
 }: {
   children?: React.ReactNode;
 }) => {
-  const projectsResult = useSubjectProjects();
-  const projectLoading = projectsResult.isLoading || !projectsResult.isFetched;
+  const { userConfiguration, isLoaded } = useSession();
+  const { user, configurations } = userConfiguration;
+  const { account } = useActiveAccount();
 
-  const useMeResult = useMe();
-  const useMeLoading = useMeResult.isLoading || !useMeResult.isFetched;
-  const { data: userData } = useUser();
-  const userId = userData?.id;
   const [selectedId, setSelectedId] = useState<string>();
-  const [previousUserId, setPreviousUserId] = useState(userId);
+  const configuration = configurations.find((c) => c.account === account?.id);
+
   const [storedProjectIdResult, setStoredProjectId, isStorageLoaded] =
-    useAsyncStorage(
-      `${selectedProjectIdKey}:${userId}`,
-      !!selectedProjectIdKey && !!userId,
-    );
+    useAsyncStorage(`${selectedProjectIdKey}:${user.id}`, isLoaded);
 
   /**
    * Initial setting of activeProject
    */
-  const hookReturnValue = useMemo<ActiveProjectProps>(() => {
-    if (
-      !userId || // wait for user id before reading and writing to storage
-      projectLoading || // wait for projects endpoint
-      useMeLoading ||
-      !isStorageLoaded
-    ) {
-      return {};
-    }
-
-    const { selectedProject, selectedSubject } = findProjectAndSubjectById(
-      selectedId ?? storedProjectIdResult,
-      projectsResult.data,
-      useMeResult.data,
-    );
-
-    if (selectedProject && selectedSubject) {
-      return {
-        activeProject: selectedProject,
-        activeSubjectId: selectedSubject.subjectId,
-        activeSubject: selectedSubject,
-      };
+  const hookReturnValue = useMemo<ActiveSubjectProject | undefined>(() => {
+    if (!isLoaded || !isStorageLoaded) {
+      return undefined;
     }
 
-    return {};
+    const projectToSelect = selectedId ?? storedProjectIdResult;
+    const subjectAndProject = projectToSelect
+      ? configuration?.subjects.find((s) => s.projectId === projectToSelect)
+      : configuration?.subjects[0];
+    return subjectAndProject ? { activeSubject: subjectAndProject } : undefined;
   }, [
-    userId,
-    projectLoading,
-    useMeLoading,
+    configuration?.subjects,
+    isLoaded,
     isStorageLoaded,
-    selectedId,
     storedProjectIdResult,
-    projectsResult.data,
-    useMeResult.data,
+    selectedId,
   ]);
 
   useEffect(() => {
-    if (hookReturnValue?.activeProject?.id) {
-      setStoredProjectId(hookReturnValue?.activeProject?.id);
-    }
-  }, [hookReturnValue?.activeProject?.id, setStoredProjectId]);
-
-  // Clear selected project when
-  // we've detected that the userId has changed
-  useEffect(() => {
```

Comment:
I assume this is handled elsewhere now.

Reply:
Since this hook no longer has anything to do with fetching subject projects, it no longer needs to keep track of the userId and force a refresh. Instead, when a session ends, either by a user logout or failure to fresh, the cached session will be discarded. This is not implemented in the code yet yet but it is on my TODO list.